reliability/stub-core/src/main/kotlin/com/datadog/android/tests/ktx/JsonObjectExt.kt

```diff
@@ -6,30 +6,103 @@
 
 package com.datadog.android.tests.ktx
 
+import com.google.gson.JsonElement
 import com.google.gson.JsonObject
 
 /**
  * A utility method to retrieve a nested attribute using the dotted notation.
  * E.g.: assuming the `this` JsonObject represents the json below, calling
- * `getString("foo.bar.spam")` will return the String `"42"`.
+ * `getString("foo.bar[0].spam")` will return the String `"lorem ipsum"`.
  *
  * {
  *   "foo": {
- *     "bar" : {
- *       "spam": "42"
- *     }
+ *     "bar" : [
+ *       {
+ *         "spam": "lorem ipsum"
+ *       }
+ *     ]
  *   }
  * }
  */
 @Suppress("UnsafeThirdPartyFunctionCall")
 fun JsonObject.getString(path: String): String? {
-    return if (has(path)) {
-        getAsJsonPrimitive(path)?.asString
+    return getAtPath(path)?.asJsonPrimitive?.asString
+}
+
+/**
+ * A utility method to retrieve a nested attribute using the dotted notation.
+ * E.g.: assuming the `this` JsonObject represents the json below, calling
+ * `getString("foo.bar[0].spam")` will return the long `42L`.
+ *
+ * {
+ *   "foo": {
+ *     "bar" : [
+ *       {
+ *         "spam": 42
+ *       }
+ *     ]
+ *   }
+ * }
+ */
+@Suppress("UnsafeThirdPartyFunctionCall")
```

Comment:
should we exclude files under `reliability` from this check?

Reply:
Yes that makes more sense indeed.